lib/migration.c

```diff
@@ -39,34 +39,106 @@
 #include "private.h"
 #include "migration_priv.h"
 
+/*
+ * This defines valid migration state transitions. Each element in the array
+ * corresponds to a FROM state and each bit of the element to a TO state. If the
+ * bit is set, then the transition is allowed.
+ * 
+ * The indices of each state are those in the vfio_user_device_mig_state enum.
+ */
+static const char transitions[8] = {
```

Comment:
VFIO_USER_DEVICE_NUM_STATES?

Reply:
Sure, but given that this is a bitmap of `char`s, if another state is added in the future stuff will still have to be changed.

Reply:
alternatively, assert that the value is 8